lib/Doctrine/DBAL/Schema/MySqlSchemaManager.php

```diff
@@ -206,6 +216,22 @@ protected function _getPortableTableColumnDefinition($tableColumn)
         return $column;
     }
 
+
+    /**
+     * Get a doctrine/mysql compatible column default from
+     * the new information_schema behaviour introduced in MariaDb >= 10.2.7.
+     *
+     * @param null|string $columnDefault default value as stored in information_schema for MariaDB >= 10.2.7
+     */
+    private function getMariaDb1027ColumnDefault(?string $columnDefault): ?string {
+        $defaultValue = ($columnDefault === 'NULL') ? null : $columnDefault;
+        if ($defaultValue !== null && ($defaultValue[0] ?? '') === "'") {
```

Comment:
The part after `&&` looks messy. If the `0` offset of the `$defaultValue` does not exist, then the condition reads as `'' === "'"`.

What if instead, we return earlier in edge cases? Like:

```php
if ($defaultValue === 'NULL') {
    return null;
}

if ($defaultValue === '') {
    return $defaultValue;
}

// process the non-empty string.